include/aspect/utilities.h

```diff
@@ -813,6 +813,18 @@ namespace aspect
          */
         std_cxx11::unique_ptr<aspect::Utilities::AsciiDataLookup<1> > lookup;
     };
+
+    /**
+     * This function computes a factor which can be used to make sure that the
+     * Jacobian remains positive definite. The factor is defined by
+     * $\frac{2\eta(\varepsilon(\bm u))}{\left[1-\frac{b:a}{\|a\| \|b\|} \right]^2\|a\|\|b\|}$,
+     * where $a=\varepsilon(\bm u)$ and $b=\frac{\partial\eta(\varepsilon(\bm u),p)}.
+     */
+    template<int dim>
+    double compute_spd_factor(const SymmetricTensor<2,dim> &strain_rate,
+                              const SymmetricTensor<2,dim> &dviscosities_dstrain_rate,
+                              const double eta);
```

Comment:
Because the viscosity appears first in the formula, I would make that the first argument of the function.